ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/Forging.hs

```diff
@@ -141,11 +141,11 @@ data BlockForging m blk = BlockForging {
       -- even when used as part of the hard fork combinator.
       --
       -- PRECONDITION: 'checkCanForge' returned @Right ()@.
-    , forgeBlock :: forall mk.
+    , forgeBlock ::
            TopLevelConfig blk
         -> BlockNo                      -- Current block number
         -> SlotNo                       -- Current slot number
-        -> TickedLedgerState blk mk     -- Current ledger state
+        -> TickedLedgerState blk EmptyMK     -- Current ledger state
```

Comment:
We do not use any state related to the UTxO here, so it might as well be `EmptyMK`. This unblocks some of the work on #344 too.